jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codemodernizer/CodeWhispererCodeModernizerSessionTest.kt

```diff
@@ -62,21 +65,27 @@ class CodeWhispererCodeModernizerSessionTest : CodeWhispererCodeModernizerTestBa
     @JvmField
     val wireMock = WireMockRule(WireMockConfiguration.wireMockConfig().dynamicPort())
 
-    lateinit var gumbyUploadUrlResponse: CreateUploadUrlResponse
-
     @Before
     override fun setup() {
         super.setup()
-        val s3endpoint = "http://127.0.0.1:${wireMock.port()}"
-        gumbyUploadUrlResponse = CreateUploadUrlResponse.builder()
-            .uploadUrl(s3endpoint)
-            .uploadId("1234")
-            .kmsKeyArn("0000000000000000000000000000000000:key/1234abcd")
-            .responseMetadata(DefaultAwsResponseMetadata.create(mapOf(ResponseMetadata.AWS_REQUEST_ID to CodeWhispererTestUtil.testRequestId)))
-            .sdkHttpResponse(
-                SdkHttpResponse.builder().headers(mapOf(CodeWhispererService.KET_SESSION_ID to listOf(CodeWhispererTestUtil.testSessionId))).build()
-            )
-            .build() as CreateUploadUrlResponse
+        ThreadLeakTracker.longRunningThreadCreated(ApplicationManager.getApplication(), "Process Proxy: Launcher")
+    }
```

Comment:
Add ThreadLeakTracker.longRunningThreadCreated() to suppress the thread leak in this unit test class.
Refer to this previous PR. https://github.com/aws/aws-toolkit-jetbrains/pull/2306

Reply:
can you share the leak stack trace? are we sure it's a leak and not a process that hasn't been cleaned up?

Reply:
here is the stack trace. Thanks so much if you can take a look.
```
java.lang.AssertionError: Thread leaked: Thread[Process Proxy: Launcher,4,main] (alive) RUNNABLE
--- its stacktrace:
 at java.base@17.0.9/sun.nio.ch.KQueue.poll(Native Method)
 at java.base@17.0.9/sun.nio.ch.KQueuePort$EventHandlerTask.poll(KQueuePort.java:200)
 at java.base@17.0.9/sun.nio.ch.KQueuePort$EventHandlerTask.run(KQueuePort.java:287)
 at java.base@17.0.9/sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:113)
 at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
 at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
 at java.base@17.0.9/java.lang.Thread.run(Thread.java:840)
---
(diagnostic: 0: sun.nio.ch.KQueue : false . poll : false 2: sun.nio.ch.KQueuePort$EventHandlerTask : false . run : false)

Leaking threads dump:
"Process Proxy: Launcher" (alive) RUNNABLE
	at java.base@17.0.9/sun.nio.ch.KQueue.poll(Native Method)
	at java.base@17.0.9/sun.nio.ch.KQueuePort$EventHandlerTask.poll(KQueuePort.java:200)
	at java.base@17.0.9/sun.nio.ch.KQueuePort$EventHandlerTask.run(KQueuePort.java:287)
	at java.base@17.0.9/sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:113)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base@17.0.9/java.lang.Thread.run(Thread.java:840)

"I/O pool 4" (alive) TIMED_WAITING
	at java.base@17.0.9/jdk.internal.misc.Unsafe.park(Native Method)
	at java.base@17.0.9/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
	at java.base@17.0.9/java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:401)
	at java.base@17.0.9/java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:903)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1061)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1122)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base@17.0.9/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base@17.0.9/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base@17.0.9/java.security.AccessController.executePrivileged(AccessController.java:776)
	at java.base@17.0.9/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base@17.0.9/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base@17.0.9/java.lang.Thread.run(Thread.java:840)

"I/O pool 5" (alive) TIMED_WAITING
	at java.base@17.0.9/jdk.internal.misc.Unsafe.park(Native Method)
	at java.base@17.0.9/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
	at java.base@17.0.9/java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:401)
	at java.base@17.0.9/java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:903)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1061)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1122)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base@17.0.9/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base@17.0.9/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base@17.0.9/java.security.AccessController.executePrivileged(AccessController.java:776)
	at java.base@17.0.9/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base@17.0.9/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base@17.0.9/java.lang.Thread.run(Thread.java:840)

"Image Fetcher 0" (dead) TERMINATED
	at java.base@17.0.9/java.lang.Object.wait(Native Method)
	at java.desktop@17.0.9/sun.awt.image.ImageFetcher.nextImage(ImageFetcher.java:154)
	at java.desktop@17.0.9/sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:207)
	at java.desktop@17.0.9/sun.awt.image.ImageFetcher.run(ImageFetcher.java:176)

"AWS-Toolkit-Metrics-Publisher" (alive) TIMED_WAITING
	at java.base@17.0.9/jdk.internal.misc.Unsafe.park(Native Method)
	at java.base@17.0.9/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
	at java.base@17.0.9/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:1672)
	at java.base@17.0.9/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1182)
	at java.base@17.0.9/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:899)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1062)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1122)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base@17.0.9/java.lang.Thread.run(Thread.java:840)

"Java2D Disposer" (alive) WAITING
	at java.base@17.0.9/java.lang.Object.wait(Native Method)
	at java.base@17.0.9/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:155)
	at java.base@17.0.9/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:176)
	at java.desktop@17.0.9/sun.java2d.Disposer.run(Disposer.java:145)
	at java.base@17.0.9/java.lang.Thread.run(Thread.java:840)


----
All other threads dump:
"PeriodicMetricReader" (alive) TIMED_WAITING
	at java.base@17.0.9/jdk.internal.misc.Unsafe.park(Native Method)
	at java.base@17.0.9/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
	at java.base@17.0.9/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:1672)
	at java.base@17.0.9/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1182)
	at java.base@17.0.9/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:899)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1062)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1122)
	at java.base@17.0.9/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base@17.0.9/java.lang.Thread.run(Thread.java:840)
......
```